llvm-runtimes/libgcc/libgcc-20.0.0.9999.ebuild

```diff
@@ -0,0 +1,169 @@
+
+		if test_compiler "${nolib_flags[@]}"; then
+			local -x LDFLAGS="${LDFLAGS} ${nolib_flags[*]}"
+			ewarn "${CC} seems to lack runtime, trying with ${nolib_flags[*]}"
+		elif test_compiler "${nolib_flags[@]}" -nostartfiles; then
+			# Avoiding -nostartfiles earlier on for bug #862540,
+			# and set available entry symbol for bug #862798.
+			nolib_flags+=( -nostartfiles -e main )
+
+			local -x LDFLAGS="${LDFLAGS} ${nolib_flags[*]}"
+			ewarn "${CC} seems to lack runtime, trying with ${nolib_flags[*]}"
+		fi
+	fi
+
+	local mycmakeargs=(
+		-DCOMPILER_RT_INSTALL_PATH="${EPREFIX}/usr/lib/clang/${LLVM_MAJOR}"
+
+		-DCOMPILER_RT_EXCLUDE_ATOMIC_BUILTIN=$(usex !atomic-builtins)
+		-DCOMPILER_RT_INCLUDE_TESTS=$(usex test)
+		-DCOMPILER_RT_BUILD_CTX_PROFILE=OFF
+		-DCOMPILER_RT_BUILD_LIBFUZZER=OFF
+		-DCOMPILER_RT_BUILD_MEMPROF=OFF
+		-DCOMPILER_RT_BUILD_ORC=OFF
+		-DCOMPILER_RT_BUILD_PROFILE=OFF
+		-DCOMPILER_RT_BUILD_SANITIZERS=OFF
+		-DCOMPILER_RT_BUILD_XRAY=OFF
+
+		-DPython3_EXECUTABLE="${PYTHON}"
+	)
+
+	if use amd64 && ! target_is_not_host; then
+		mycmakeargs+=(
+			-DCAN_TARGET_i386=$(usex abi_x86_32)
+			-DCAN_TARGET_x86_64=$(usex abi_x86_64)
+		)
+	fi
+
+	if use prefix && [[ "${CHOST}" == *-darwin* ]] ; then
+		mycmakeargs+=(
+			# setting -isysroot is disabled with compiler-rt-prefix-paths.patch
+			# this allows adding arm64 support using SDK in EPREFIX
+			-DDARWIN_macosx_CACHED_SYSROOT="${EPREFIX}/MacOSX.sdk"
+			# Set version based on the SDK in EPREFIX.
+			# This disables i386 for SDK >= 10.15
+			-DDARWIN_macosx_OVERRIDE_SDK_VERSION="$(realpath ${EPREFIX}/MacOSX.sdk | sed -e 's/.*MacOSX\(.*\)\.sdk/\1/')"
+			# Use our libtool instead of looking it up with xcrun
+			-DCMAKE_LIBTOOL="${EPREFIX}/usr/bin/${CHOST}-libtool"
+		)
+	fi
+
+	if use test; then
+		mycmakeargs+=(
+			-DLLVM_EXTERNAL_LIT="${EPREFIX}/usr/bin/lit"
+			-DLLVM_LIT_ARGS="$(get_lit_flags)"
+
+			-DCOMPILER_RT_TEST_COMPILER="${EPREFIX}/usr/lib/llvm/${LLVM_MAJOR}/bin/clang"
+			-DCOMPILER_RT_TEST_CXX_COMPILER="${EPREFIX}/usr/lib/llvm/${LLVM_MAJOR}/bin/clang++"
+		)
+	fi
+
+	cmake_src_configure
+}
+
+src_compile() {
+	local -x CC=${CTARGET}-clang CXX=${CTARGET}-clang++
+	local libdir=$(get_libdir)
+
+	cmake_src_compile
+
+	$(tc-getCPP) \
+		"${WORKDIR}/llvm-libgcc/gcc_s.ver.in" \
+		-o gcc_s.ver || die
+	$(tc-getCC) -nostdlib \
+		${LDFLAGS} \
+		-Wl,--version-script,gcc_s.ver \
+		-Wl,--whole-archive \
+		"${BUILD_DIR}/lib/linux/libclang_rt.builtins-${CTARGET%%-*}.a" \
+		"${EPREFIX}/usr/${libdir}/libunwind.a" \
+		-Wl,-soname,libgcc_s.so.1.0 \
+		-lc -shared \
+		-o libgcc_s.so.1.0 || die
+}
+
+src_install() {
+	local libdir=$(get_libdir)
+	dolib.so libgcc_s.so.1.0
+	insinto "/usr/${libdir}"
+	newins "${BUILD_DIR}/lib/linux/libclang_rt.builtins-${CTARGET%%-*}.a" libgcc.a
+	dosym libgcc_s.so.1.0 "/usr/${libdir}/libgcc_s.so.1"
+	dosym libgcc_s.so.1 "/usr/${libdir}/libgcc_s.so"
+	dosym libunwind.a "/usr/${libdir}/libgcc_eh.a"
+}
+
+src_test() {
```

Comment:
Please keep phases sorted in run order, i.e. move `src_test` before `src_install`.